src/toil/test/src/importExportFileTest.py

```diff
@@ -16,62 +16,83 @@
-        with open(self.srcFile, 'w') as f:
-            f.write(os.urandom(2**10))
-
-    def tearDown(self):
-        shutil.rmtree(self.srcFile.rsplit('/', 1)[0])
+        super(ImportExportFileTest, self).setUp()
+        self._tempDir = self._createTempDir()
+        self.dstFile = '%s/%s' % (self._tempDir, 'out')
 
-    def testImportExportFile(self):
+    def _importExportFile(self, options, fail):
+        with Toil(options) as toil:
+            if not options.restart:
+                # stage files
+                srcFile = '%s/%s%s' % (self._tempDir, 'in', uuid.uuid4())
+                with open(srcFile, 'w') as f:
+                    f.write('Hello')
+                inputFileID = toil.importFile('file://' + srcFile)
+
+                # write fail value and sets self.failFileID to its ID
+                # this ID will be passed to the job. The boolean value
+                # written to it determines if the job fails or not.
+                with toil._jobStore.writeFileStream() as (failToggle, failFileID):
+                    self.failFileID = failFileID
+                    failToggle.write(str(fail))
+
+                # run workflow
+                outputFileID = toil.start(HelloWorld(inputFileID, self.failFileID))
+            else:
+                # write fail value
+                # see above comment for explanation
+                with toil._jobStore.updateFileStream(self.failFileID) as failToggle:
+                    failToggle.write('False')
+
+                # run workflow
+                outputFileID = toil.restart()
+
+            # export file to dstFile
+            toil.exportFile(outputFileID, 'file://' + self.dstFile)
+            with open(self.dstFile, 'r') as f:
+                assert f.read() == "HelloWorld!"
+
+
+    def _importExport(self, restart):
+        # create options for workflow
         options = Job.Runner.getDefaultOptions(self._getTestJobStorePath())
         options.logLevel = "INFO"
-        job = HelloWorld()
-        with Toil(options) as toil:
-            jobStoreFileID = toil.importFile('file://%s' % self.srcFile)
-            assert toil.jobStore.fileExists(jobStoreFileID)
-
-            destPath = '%s/%s' % (self.srcFile.rsplit('/', 1)[0], uuid.uuid4())
-            toil.exportFile(jobStoreFileID, 'file://%s' % destPath)
-            assert os.path.exists(destPath)
 
-            toil.run(job)
+        if restart:
+            try:
+                self._importExportFile(options, fail=True)
+            except FailedJobsException:
+                options.restart = True
 
+        self._importExportFile(options, fail=False)
 
-class HelloWorld(Job):
-    def __init__(self):
-        Job.__init__(self,  memory=100000, cores=2, disk="3G")
-
-    def run(self, fileStore):
-        fileID = self.addChildJobFn(childFn, cores=1, memory="1M", disk="1M").rv()
-        self.addFollowOn(FollowOn(fileID))
+    @classmethod
+    def make(cls):
```

Comment:
I think with just two cases, it will be hard to motivate the use of generative programming. Just write these two test methods by hand so you don't have to explain too much when you demo this on Friday.